### PR TITLE
Adding missing instructions to Contributing.md

### PR DESCRIPTION
**Pull request by Nutomic**
See #930.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -67,6 +68,21 @@ Enter password for new role: peertube
...
+```
+
+Then enable extensions PeerTube needs:
+
+```
+$ sudo -u postgres psql -c "CREATE EXTENSION pg_trgm;" peertube_dev
+$ sudo -u postgres psql -c "CREATE EXTENSION unaccent;" peertube_dev
+```
+
 In dev mode, administrator username is **root** and password is **test**.
```
Comment (Jorropo) [inline .github/CONTRIBUTING.md]: You should remove stars cause if like me you read this doc in a basic text editor you will be confused about the password.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -52,6 +52,7 @@ Then clone the sources and install node modules:
 
 ```
 $ git clone https://github.com/Chocobozzz/PeerTube
+$ git checkout develop
```
Comment (Jorropo) [inline .github/CONTRIBUTING.md]: You don't need to checkout develop, if what you are testing is in an other branch you checkout the other branch.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -67,6 +68,21 @@ Enter password for new role: peertube
 # sudo -u postgres createdb -O peertube peertube_dev
 ```
 
+You also need to create a system user for Peertube:
```
Comment (Chocobozzz) [inline .github/CONTRIBUTING.md]: Why did you need this peertube user? You should be able to dev on peertube with your regular user

Comment (Nutomic) [reply .github/CONTRIBUTING.md]: Ah I misunderstood something, will remove this.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (Nutomic) [reply .github/CONTRIBUTING.md]: Right, I'm adding a comment regarding this.

**Comment by Nutomic** [comment]
Updated PR.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -56,6 +56,10 @@ $ cd PeerTube
 $ yarn install --pure-lockfile
 ```
 
+Note that development is done on the `develop` branch. If you want to hack on
```
Comment (Jorropo) [inline .github/CONTRIBUTING.md]: Develop isn't really made on develop branch, in fact you should durring work on an other branch and merge into develop when finish.

Comment (Chocobozzz) [reply .github/CONTRIBUTING.md]: I agree but to keep it as simple as possible, we don't really want to specify how the developer should work. If they want to use the git flow, they can, if they want to work directly on develop on their fork, they can too.

**Comment by Chocobozzz** [comment]
Thanks @Nutomic
